cdap-ui/server/express.js

```diff
@@ -180,6 +180,22 @@ function makeApp(authAddress, cdapConfig, uiSettings) {
   app.use(bodyParser.urlencoded({ extended: false }));
   app.use(cookieParser());
 
+  if (!isModeDevelopment()) {
+    let marketUrl = url.parse(cdapConfig['market.base.url']);
```

Comment:
is this needed? wouldn't the url already contains the protocol and host?

Reply:
It also has a `v2/dev` which was not working. CSP only requires the domain.